### PR TITLE
Fixing the rolebinding service accounts

### PR DESCRIPTION
**Pull request by Shubham-astro**
## Description

This PR intends to make sure that the service account name picked by the rolebinding template is similar to service account name when when name is passed using astronomer.configSyncer.serviceaccount.name to "test1" with serviceaccount.create set to true

## Related Issues

Do not merge this PR until this text is replaced with links to related issues.

Related astronomer/issues#6734

## Testing

Test needs to be done with serviceaccount.name and serviceaccount.create set to true 

## Merging

cherry-pick to 0.36

### CODE REVIEW

File: charts/astronomer/templates/commander/commander-rolebinding.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@
 {{- $useClusterRoles := and .Values.global.rbacEnabled .Values.global.clusterRoles (not .Values.global.features.namespacePools.enabled) }}
 {{- $shouldCreateResources := and .Values.global.rbacEnabled (or (and .Values.global.clusterRoles (not .Values.global.features.namespacePools.enabled )) $namespacepoolsRBACresources) }}
 {{- $namespaces := .Values.global.features.namespacePools.namespaces.names }}
+{{- $data := ( include "commander.ServiceAccount" . )  -}}
```
Comment (pgvishnuram) [inline charts/astronomer/templates/commander/commander-rolebinding.yaml:12]: change this to commanderServiceAccount as variable declaration

File: charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@
 {{- $useClusterRoles := and .Values.global.rbacEnabled (not .Values.global.features.namespacePools.enabled)}}
 {{- $namespaces := .Values.global.features.namespacePools.namespaces.names }}
 {{- $shouldCreateResources := and .Values.global.rbacEnabled (or (and .Values.global.clusterRoles (not .Values.global.features.namespacePools.enabled )) $namespacepoolsRBACresources) }}
+{{- $data := ( include "configSyncer.ServiceAccount" . )  -}}
```
Comment (pgvishnuram) [inline charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml:13]: change this to configSyncerServiceAccountName  as variable declaration

Comment (Shubham-astro) [reply charts/astronomer/templates/config-syncer/config-syncer-rolebinding.yaml]: done

File: charts/astronomer/templates/commander/commander-rolebinding.yaml (hunk 1 above)
Comment (Shubham-astro) [reply charts/astronomer/templates/commander/commander-rolebinding.yaml]: done

**Comment by pgvishnuram** [comment]
can you check for kube-state-metrics as well

**Comment by Shubham-astro** [comment]
> astronomer/issues#6734

done and pushed changes for all components

File: charts/astronomer/templates/houston/api/houston-bootstrap-rolebinding.yaml [hunk 3]
```diff
@@ -21,6 +21,6 @@ roleRef:
   name: {{ .Release.Name }}-houston-bootstrapper
 subjects:
   - kind: ServiceAccount
-    name: {{ template "houston.bootstrapperServiceAccount" . }}
+    name: {{ template  "houston.bootstrapperServiceAccount" . }}
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/api/houston-bootstrap-rolebinding.yaml:24]:
```suggestion
    name: {{ template "houston.bootstrapperServiceAccount" . }}
```

**Review by danielhoherd — APPROVED** [review]
Looks good. Great use of the `set` data type in the tests.

**Comment by pgvishnuram** [comment]
grafana service account changes were missing that has been added and merged back to this PR

**Review by pgvishnuram — APPROVED** [review]

**Pull request merged**